### PR TITLE
Typing updates.

### PR DESCRIPTION
**Pull request by dzearing**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
-    "autoprefixer": "6.3.4",
+    "autoprefixer": "6.3.6",
     "gulp": "~3.9.1",
     "gulp-changed": "1.3.0",
     "gulp-clean-css": "2.0.4",
-    "gulp-clone": "^1.0.0",
-    "gulp-core-build": "~0.4.7",
+    "gulp-clip-empty-files": "0.1.1",
+    "gulp-clone": "1.0.0",
+    "gulp-core-build": "~0.6.0",
```
Comment (iclanton) [inline package.json]: 0.6.1 is the latest. I think we should try to stay in as tight lockstep with GCB as possible for these.

File: src/SassTask.ts [hunk 2]
```diff
@@ -27,56 +28,67 @@ export class SassTask extends GulpTask<ISassTaskConfig> {
     'src/**/*.scss.ts'
   ];
 
-  public executeTask(gulp, completeCallback): any {
+  public executeTask(
+    gulp: gulp.Gulp,
+    completeCallback?: (result?: any) => void
+  ): Promise<any> | NodeJS.ReadWriteStream | void {
```
Comment (iclanton) [inline src/SassTask.ts]: This should just be NodeJS.ReadWriteStream.

File: src/SassTask.ts [hunk 3]
```diff
@@ -27,56 +28,67 @@ export class SassTask extends GulpTask<ISassTaskConfig> {
...
-                   this.processFiles(gulp, merge, moduleSrcPattern, completeCallback, modulePostCssPlugins));
+      return merge(this._processFiles(gulp, merge, srcPattern, completeCallback, postCSSPlugins),
+                   this._processFiles(gulp, merge, moduleSrcPattern, completeCallback, modulePostCssPlugins));
     }
   }
 
-  private processFiles(gulp, merge, srcPattern, completeCallback, postCSSPlugins): NodeJS.ReadWriteStream {
+  private _processFiles(
+    gulp: gulp.Gulp,
+    merge: any,
```
Comment (iclanton) [inline src/SassTask.ts]: Why not just make this global? I understand not loading it if it isn't needed, but it's a very small package and I'd rather we have typesafety and elimiate the "any".

If we don't load it globally, can we at least change this type to "(...stream: NodeJS.ReadWriteStream[]): NodeJS.ReadWriteStream" so we at least get some typesafety?

File: src/SassTask.ts [hunk 4]
```diff
@@ -27,56 +28,67 @@ export class SassTask extends GulpTask<ISassTaskConfig> {
     'src/**/*.scss.ts'
   ];
 
-  public executeTask(gulp, completeCallback): any {
+  public executeTask(
+    gulp: gulp.Gulp,
+    completeCallback?: (result?: any) => void
```
Comment (iclanton) [inline src/SassTask.ts]: Realistically - do we ever pass a non-string into completeCallback? Maybe we should change that type.

**Comment by iclanton** [comment]
Approved, with suggestions.

File: src/SassTask.ts (hunk 3 above)
Comment (dzearing) [inline src/SassTask.ts]: fixed.

File: src/SassTask.ts (hunk 4 above)
Comment (dzearing) [inline src/SassTask.ts]: fixed.

**Pull request merged**